### PR TITLE
Add rust language guide

### PR DESCRIPTION
**Pull request by craig-osterhout**
### Proposed changes

Add language-specific guide for Rust using docker init.
https://deploy-preview-17610--docsdocker.netlify.app/language/rust/

### Related issues (optional)
ENGDOCS-1358

### CODE REVIEW

**Comment by sfavello** [comment]
🤩 , This looks really awesome! Super excited for the Rust language guide to be out in the wild.

**Review by stephaurelio — COMMENTED** [review]
Left a few comments but looks great so far ✨ I'll keep an eye on this PR in case you make any of the suggested changes so I can check back in to approve!

File: language/rust/build-images.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+You should see output similar to the following.
+
+```console
+Hello, Docker!
+```
+
+## Create a Dockerfile for Rust
+
+Now that you have an application, you can use `docker init` to create a Dockerfile for it. Inside the `docker-rust-app` directory, run the `docker init` command. Refer to the following example to answer the prompts from `docker init`.
```
[the remark this answers is not in the document]
Comment (craig-osterhout) [reply language/rust/build-images.md:57]: Yep! Soon™ after evaluating community feedback on this one.

[the remark this answers is not in the document]
Comment (cmrigney) [reply language/rust/build-images.md:57]: Nice 👍

**Review by stephaurelio — APPROVED** [review]
These changes LGTM ✨

**Pull request merged**